docs/content/en/docs/yaml-crd-ref/taskdefinition.md

```diff
@@ -3,3 +3,310 @@ title: KeptnTaskDefinition
 description: Define tasks that can be run pre- or post-deployment
 weight: 89
 ---
+
+
+A `KeptnTaskDefinition` defines tasks
+that are run by the Keptn Lifecycle Toolkit
+as part of the pre- and post-deployment phases of a
+[KeptnApp](./app.md).
+
+## Yaml Synopsis
+
+```yaml
+apiVersion: lifecycle.keptn.sh/v?alpha?
+kind: KeptnTaskDefinition
+metadata:
+  name: <task-name>
+spec:
+  function:
+    inline | httpRef | functionRef:
+    [parameters:
+      map:
+        textMessage: "This is my configuration"]
+    [secureParameters:
+      secret: slack-token]
+```
+
+## Fields
+
+* **apiVersion** -- API version being used.
+`
+* **kind** -- Resource type.
+   Must be set to `KeptnTaskDefinition`
+
+* **metadata**
+  * **name** -- Unique name of this task.
+    Names must comply with the
+    [Kubernetes Object Names and IDs](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names)
+    specification.
+
+* **spec**
+  * **function** -- Code to be executed,
+    expressed as a [Deno](https://deno.land/) script.
+    Refer to [function runtime](https://github.com/keptn/lifecycle-toolkit/tree/main/functions-runtime)
+    for more information about the runtime.
+
+    The `function` can be defined as one of the following:
+
+    * **inline** - Include the actual executable code to execute.
+      This can be written as a full-fledged Deno script
+      that is included in this file.
+      For example:
+
+      ```yaml
+      function:
+        inline:
+          code: |
+            console.log("Deployment Task has been executed");
+      ```
+
+    * **httpRef** - Specify a Deno script to be executed at runtime
+      from the remote webserver that is specified.
+      For example:
+
+      ```yaml
+      name: hello-keptn-http
+        spec:
+            function:
+              httpRef:
+                url: "https://www.example.com/yourscript.js"
+      ```
+
+      Note that the file referenced is actually JavaScript,
```

Comment:
I think we can leave out this paragraph to avoid confusion, since Deno has tighter restrictions when it comes to permissions or importing other libraries. So it might be the case that a script working in e.g. nodeJS will not function properly in Deno

Reply:
Ugh!  We have the example that is calling JavaScript and I notice that Andi and others are referring to this as JavaScript.  And we don't say anything about Deno having tighter restrictions.  Let me do a bit of a rewrite here.

Reply:
I removed this sentence and added a paragraph about Deno execution in the "Usage" section.